src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java

```diff
@@ -238,98 +274,135 @@ private synchronized void syncAcl() {
 		logger.debug("Syncing the ACL to ElasticSearch");
 		try {
 			logger.debug("Loading SearchGuard ACL...");
-			final SearchGuardACL acl = loadAcl(esClient);
+
+			SearchGuardRoles roles = readRolesACL(esClient);
+			SearchGuardRolesMapping rolesMapping = readRolesMappingACL(esClient);
+			
 			logger.debug("Syncing from cache to ACL...");
-			acl.syncFrom(cache, userProfilePrefix);
-			write(esClient, acl);
+			roles.syncFrom(cache, userProfilePrefix);
+			rolesMapping.syncFrom(cache, userProfilePrefix);
+			
+			writeACL(esClient, roles, rolesMapping);
 		} catch (Exception e) {
 			logger.error("Exception while syncing ACL with cache", e);
 		}
 	}
 	
-	private SearchGuardACL loadAcl(Client esClient) throws IOException {
-		GetRequest request = esClient.prepareGet(searchGuardIndex, SEARCHGUARD_TYPE, SEARCHGUARD_ID)
+	// This will be the new way to do this?
+	private SearchGuardRoles readRolesACL(Client esClient) throws IOException {
+		
+		GetRequest getRequest = esClient.prepareGet("searchguard", "roles", "0")
 				.setRefresh(true).request();
-		request.putInContext(OS_ES_REQ_ID, ACL_FILTER_ID);
-		GetResponse response = esClient.get(request).actionGet(); // need to worry about timeout?
-		return mapper.readValue(response.getSourceAsBytes(), SearchGuardACL.class);
+		GetResponse response = esClient.get(getRequest).actionGet(); // need to worry about timeout?
```

Comment:
We probably will only be able to answer your question based on observation.  The previous impl utilized the lock and notifier because of the 'stop the world' threading on incoming requests.  I think the approach we took was to make a best effort to get what we wanted and then defer to what was cached and/or most restrictive.